qiskit/algorithms/optimizers/optimizer.py

```diff
@@ -181,6 +190,7 @@ def __init__(self):
         self._initial_point_support_level = self.get_support_level()["initial_point"]
         self._options = {}
         self._max_evals_grouped = 1
+        self.callback = callback
```

Comment:
In the past we have always done private instance vars and getter/setter properties to expose them as the properties get documented this way which the do not when they are simple instance vars like this. Recently we have started to use Attributes to document these - but this is for new algorithms that we have just introduced (eg PVQD in evolvers). Here for consistency, and since its only one variable, I think a getter/setter property.